server/supernova/server/node_graph.cpp

```diff
@@ -72,7 +72,7 @@ int32_t node_graph::generate_node_id(void) {
     int32_t new_id;
     do {
         for (;;) {
-            new_id = -std::abs<int32_t>(server_node::hash(generated_id));
+            new_id = -std::abs(static_cast<int32_t>(server_node::hash(generated_id)));
```

Comment:
it looks to me like both of these casts are unnecessary, since the functions already return `int32_t`.

Reply:
Without this cast I'm getting `ambiguous call to overloaded function` error:
<details>

```shell
node_graph.cpp
Info: Boost.Config is older than your compiler version - probab
ly nothing bad will happen - but you may wish to look for an up
date Boost version.  Define BOOST_CONFIG_SUPPRESS_OUTDATED_MESS
AGE to suppress this message.
C:\Users\dyfee\Documents\src\supercollider\server\supernova\dsp_t
hread_queue/dsp_thread_queue.hpp(70): warning C4200: nonstandard
extension used: zero-sized array in struct/union [C:\Users\dyfee\
Documents\src\supercollider\build\server\supernova\libsupernova.v
cxproj]
C:\Users\dyfee\Documents\src\supercollider\server\supernova\dsp
_thread_queue/dsp_thread_queue.hpp(70): note: This member will
be ignored by a defaulted constructor or copy/move assignment o
perator
c:\users\dyfee\documents\src\supercollider\server\supernova\ser
ver\dependency_graph_generator.hpp(191): note: see reference to
 class template instantiation 'nova::dsp_thread_queue_item<nova
::dsp_queue_node<nova::rt_pool_allocator<void *>>,nova::rt_pool
_allocator<void *>>::successor_list::data_t' being compiled
C:\Users\dyfee\Documents\src\supercollider\server\supernova\serve
r\node_graph.cpp(62): error C2668: 'abs': ambiguous call to overl
oaded function [C:\Users\dyfee\Documents\src\supercollider\build\
server\supernova\libsupernova.vcxproj]
C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucr
t\stdlib.h(289): note: could be 'int abs(int)'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucr
t\stdlib.h(359): note: or       'long abs(const long) throw()'
C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucr
t\stdlib.h(364): note: or       '__int64 abs(const __int64) thr
ow()'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\V
C\Tools\MSVC\14.16.27023\include\cstdlib(19): note: or       'd
ouble abs(double) noexcept'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\V
C\Tools\MSVC\14.16.27023\include\cstdlib(24): note: or       'f
loat abs(float) noexcept'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\V
C\Tools\MSVC\14.16.27023\include\cstdlib(29): note: or       'l
ong double abs(long double) noexcept'
C:\Users\dyfee\Documents\src\supercollider\server\supernova\ser
ver\node_graph.cpp(62): note: while trying to match the argumen
t list '(size_t)'
Done Building Project "C:\Users\dyfee\Documents\src\supercollider
\build\server\supernova\libsupernova.vcxproj" (default targets) -
- FAILED.
```
</details>